include/testharness/TestHarness.h

```diff
@@ -54,14 +56,21 @@ class TestHarness {
 
 private:
   // The results of the tests.
+  //  NOTE we keep both a result manager and
+  //  the result in the TestSet to ensure in-ordre
+  //  printing
   ResultManager results;
 
 private:
+  // thread control
+  void spawnThreads();
+
   // test running
-  bool runTestsForToolChain(std::string tcId, std::string exeName);
+  void threadRunTestsForToolChain(std::string tcId, std::string exeName, std::reference_wrapper<SubPackage> subpackage);
```

Comment:
Why is this `std::reference_wrapper` and not just `SubPackage&`? Typical use for std::reference_wrapper is for storage in containers or template shenanigans.